modules/core/src/main/java/io/fluo/recipes/map/CollisionFreeMap.java

```diff
@@ -168,12 +185,16 @@ public U apply(Bytes b) {
         } else {
           tx.set(currentValueRow, DATA_COLUMN, newVal);
         }
-        V cv = null;
-        if (currVal != null) {
-          // deserialize again in case combiner mutated Object
-          cv = serializer.deserialize(currVal.toArray(), valType);
+
+        if (partiallyProcessedKey == null || !partiallyProcessedKey.equals(entry.getKey())) {
+          V cv = null;
+          if (currVal != null) {
+            // deserialize again in case combiner mutated Object
+            cv = serializer.deserialize(currVal.toArray(), valType);
+          }
+
+          updatesToReport.add(new Update<K, V>(kd, cv, nv.orNull()));
```

Comment:
I have been looking into this.  There are two guarantees I want the CFM to offer.  

 * When update observer is called, all updates committed before are combined.
 * When update observer is called, whats passed in as the old value will be the previous new value.  

Need to document these guarantees.  In order to maintain these two guarantees and respect a memory limit, I think I need to combine some updates and push the result of that back into the update queue.  However the current typing prevents this because the combiner output type differs from the update type.  I am thinking of making them the same type so I can combine a subset of updates and push them back as an update.